Name: Share Link Message Option
Version: 1.3.5-1
Author: Fritos1406
Description: Adds An option to share link through messaging app

--- .orig/usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
+++ /usr/palm/applications/com.palm.app.browser/app/controllers/page-assistant.js
@@ -625,7 +625,8 @@
 
 			var popupItems = [
 				{label: $L('Open In New Card'), command:'openNew'},
-				{label: $L('Share Link'), command:'shareUrl'},
+				{label: $L('Share Link Email'), command:'shareUrl'},
+				{label: $L('Share Link Message'), command:'shareUrlMms'},
 				{label: $L('Copy URL'), command:'copyUrl'},
 				{label: $L('Copy to Photos'), command:'copyToPhotos'},
 				{label: $L('Share Image'), command:'shareImage'},
@@ -675,6 +676,9 @@
 					case 'shareUrl':
 						this._shareUrl(urlInfo.url, urlInfo.desc, false /*no capture*/);
 						break;
+					case 'shareUrlMms':
+						this._shareUrlMms(urlInfo.url, urlInfo.desc, false /*no capture*/);
+						break;						
 					case 'copyUrl':
 						this.controller.stageController.setClipboard(urlInfo.url);
 						break;
@@ -697,6 +701,7 @@
 				if (urlInfo.url) {
 					usedItems.push( findItem('openNew') );
 					usedItems.push( findItem('shareUrl') );
+					usedItems.push( findItem('shareUrlMms') );
 					usedItems.push( findItem('copyUrl') );
 				}
 
@@ -1763,6 +1768,45 @@
 		method: 'open',
 		parameters: parameters
 	});
+};PageAssistant.prototype._shareUrlMms = function(url, title, withCapture) {
+
+	if (url === undefined) {
+		return;
+	}
+
+	if (!title) {
+		try {
+			title = $L("page at #{host}").interpolate({host: UrlUtil.getUrlHost(url)});
+		}
+		catch (e) {
+			title = url;
+		}
+	}
+
+	var captureFile;
+	if (withCapture) {
+		captureFile = '/tmp/captures/browser_page.png';
+
+		this._webView.mojo.saveViewToFile(captureFile);
+	}
+
+	var msg = $L("Here's a website I think you'll like: #{src}").interpolate(
+			{src: url, title: title});
+	var parameters = {
+		id: 'com.palm.app.messaging',
+		params: {
+			messageText: msg
+		}
+	};
+
+	if (withCapture) {
+		parameters.params.attachments = [{fullPath: captureFile}];
+	}
+
+	this.controller.serviceRequest('palm://com.palm.applicationManager', {
+		method: 'open',
+		parameters: parameters
+	});
 };
 
 PageAssistant.prototype._setWallpaper = function(pathToImage) {
